src/stores/proposalManagement.js

```diff
@@ -0,0 +1,53 @@
+import { defineStore } from 'pinia'
+import { fetchApiCall } from '../utils/api'
+import { useConfigurationStore } from './configuration'
+
+export const useProposalStore = defineStore('proposalManagement', {
+  state () {
+    return {
+      proposalsWithRealTimeAllocation: [],
+      proposalsWithNormalTimeAllocation: [],
+      allActiveProposals: []
+    }
+  },
+  persist: true,
+  actions: {
+    sortProposals (proposals) {
+      this.allActiveProposals = []
+      this.proposalsWithRealTimeAllocation = []
+      this.proposalsWithNormalTimeAllocation = []
+
+      const activeProposals = proposals.filter(proposal => proposal.current === true)
+      this.allActiveProposals = activeProposals
+      for (const proposal of activeProposals) {
+        // Check if the proposal has any realtime allocation with available credit
+        const hasRealTimeAllocation = proposal.time_allocations.some(time => {
+          const realTimeAllocation = time.realtime_allocation
+          const realTimeUsed = time.realtime_time_used
+          return realTimeAllocation - realTimeUsed > 0
+        })
+        const hasNormalTimeAllocation = proposal.time_allocations.some(time => {
+          const normalTimeAllocation = time.std_allocation
+          const normalTimeUsed = time.std_time_used
+          return normalTimeAllocation - normalTimeUsed > 0
+        })
+        if (hasRealTimeAllocation) {
+          this.proposalsWithRealTimeAllocation.push(proposal)
+        }
+        if (hasNormalTimeAllocation) {
+          this.proposalsWithNormalTimeAllocation.push(proposal)
+        }
+      }
+    },
+    async fetchProposals () {
+      const configurationStore = useConfigurationStore()
```

Comment:
I added this to the /profile/ endpoint in case you wanted to get it once when loading the users profile (on login). If you want to call this many times, it's probably more efficient to just call the /proposals/ endpoint for the user, and then also filter the query with active=True and semester= the current semester...